### PR TITLE
Allow users to set a subfolder as the configuration root

### PR DESCRIPTION
**Pull request by PedroLamas**
Following up on #79 and #80, this PR introduces a new setting property to the extension that will allow users to set the relative path for their Home Assistant configuration root folder.

The `configRootPath` settings and it is optional string, so this is not a breaking change.

Users will normally only use this setting on the ".vscode/settings.json" file of the workspace, but they can obviously change the global setting without any problem at all.

The `discoverFilesAndUpdateSchemas` method is not called when the extension loads, but instead whenever the plugin settings load/change.

### CODE REVIEW

**Comment by PedroLamas** [comment]
Anything I can do to help this getting approved?

**Comment by keesschollaart81** [comment]
> Anything I can do to help this getting approved?

Sorry, I had a review pending, I did not know I had to 'submit' such a review (GitHub newbie here ;-) )

File: src/server/server.ts [hunk 1]
```diff
@@ -113,10 +113,10 @@ connection.onInitialize(async params => {
     if (!configurationService.isConfigured) {
       connection.sendNotification("no-config");
     }
-  });
 
-  //fire and forget
-  setTimeout(discoverFilesAndUpdateSchemas, 0);
+    //fire and forget
+    setTimeout(discoverFilesAndUpdateSchemas, 0);
+  });
```
[the remark this answers is not in the document]
Comment (PedroLamas) [reply src/server/server.ts]: That was actually on purpose, following what I wrote on the PR description:

> The discoverFilesAndUpdateSchemas method is not called when the extension loads, but instead whenever the plugin settings load/change.

We need to call that method later, once the settings are loaded. From my tests, there are no side-effects on this change.

**Comment by keesschollaart81** [comment]
I updated your code with bf83ac2fe3b1c62529e595e31c069af0421c11eb since I figured I could detect the subfolder automatically, which makes it even more convenient for users.

**Comment by PedroLamas** [comment]
That's great! I didn't take that approach as I was under the impression that you wanted to avoid having to search the whole tree, but I'm all in for automatic discovery!

**Pull request merged**